src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -15,17 +15,20 @@
 #include "gcs_node_manager.h"
 #include <ray/common/ray_config.h>
 #include <ray/gcs/pb_util.h>
+#include <ray/gcs/redis_client.h>
 #include <ray/protobuf/gcs.pb.h>
 
 namespace ray {
 namespace gcs {
 GcsNodeManager::GcsNodeManager(boost::asio::io_service &io_service,
                                gcs::NodeInfoAccessor &node_info_accessor,
-                               gcs::ErrorInfoAccessor &error_info_accessor)
+                               gcs::ErrorInfoAccessor &error_info_accessor,
+                               std::shared_ptr<gcs::RedisClient> redis_client)
```

Comment:
Don't rely on `RedisClient` directly here. You could replace it with `gcs::GcsPubSub`.

Reply:
This is a good idea, i have fix it.